indigo_social/forms.py

```diff
@@ -50,7 +49,14 @@ def clean_username(self):
         return username
 
     def save(self, commit=True):
+        if self.cleaned_data.get('new_profile_photo'):
+            self.instance.profile_photo = self.cleaned_data['new_profile_photo']
```

Comment:
This is the perfect point to resize the photo, rather than doing it on every save. At this point we know it's new, we know it must be resized, and we know it exists.

Reply:
We can also resize it based on the current in-memory image file, rather than writing to s3, reading back, resizing, writing back to s3.